### ISSUE TITLE
Hard coded segmentMimiTypes, why?

### ISSUE THREAD

**Issue by sberryman** [kanongil/node-hls-segment-reader#8]
https://github.com/kanongil/node-hls-segment-reader/blob/86499532f115a17b6ba98bd6209a1cd1c45ae090/lib/segment-reader.js#L32-L47

You have hard coded that array and I've run into a case where I have a MIME type of `application/octet-stream`. I've also come across MIME type of `application/binary`

Thoughts on allowing a user to override or push values into the set?

**Comment by kanongil** [comment]
Yes. Currently, it can be "hacked" by subclassing the reader, and providing your own `segmentMimeTypes()` getter.

I'm open to adding an actual option (probably as a function that is called to verify the mime type).

**Comment by sberryman** [comment]
I'm open for a function to verify the type. Thanks for the great code though, it is the perfect way for me to dump HLS streams to disk. So glad you posted the decrypt code https://github.com/kanongil/node-hls-tools/blob/master/lib/segment-decrypt.js in that project as well.

**Comment by ox-michaelradionov** [comment]
For anyone interested in adding an extra mime type using a "hack" provided by @kanongil above, here is an implementation

```
const HLSSegmentReader = require('hls-segment-reader');

class ExtendedHLSSegmentReader extends HLSSegmentReader {

  get segmentMimeTypes() {
    const extendedMimeTypes = new Set(super.segmentMimeTypes);
    extendedMimeTypes.add('application/octet-stream');
    return extendedMimeTypes;
  }

}

const reader = new ExtendedHLSSegmentReader('foo.m3u8');
```
